kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -145,6 +149,7 @@ public void commit() throws IOException {
     }
 
     try {
+      compressionType.finalize(compressionFilter);
```

Comment:
Although it should never be called before a `wrapForCompression` is called first (according to the pattern that's intended for this functionality), we should make sure that any implementation of `finalize` behaves correctly on `null` (doesn't through `NPE`).

Reply:
Adding a note on this to the javadoc for `finalize`. The one existing override of that method already handles nulls, though, as they'll fail the instanceof check and trigger a ConnectException to be thrown.